packages/media-parser/src/internal-parse-media.ts

```diff
@@ -0,0 +1,327 @@
+import {emitAvailableInfo} from './emit-available-info';
+import {getFieldsFromCallback} from './get-fields-from-callbacks';
+import {getAvailableInfo, hasAllInfo} from './has-all-info';
+import {Log} from './log';
+import type {
+	AllParseMediaFields,
+	InternalParseMedia,
+	InternalParseMediaOptions,
+	Options,
+	ParseMediaCallbacks,
+	ParseMediaFields,
+	ParseMediaResult,
+} from './options';
+import {performSeek} from './perform-seek';
+import {runParseIteration} from './run-parse-iteration';
+import {makeParserState} from './state/parser-state';
+import {throttledStateUpdate} from './throttled-progress';
+
+export const internalParseMedia: InternalParseMedia = async function <
+	F extends Options<ParseMediaFields>,
+>({
+	src,
+	fields: _fieldsInReturnValue,
+	reader: readerInterface,
+	onAudioTrack,
+	onVideoTrack,
+	signal,
+	logLevel,
+	onParseProgress: onParseProgressDoNotCallDirectly,
+	progressIntervalInMs,
+	mode,
+	onDiscardedData,
+	onError,
+	...more
+}: InternalParseMediaOptions<F>) {
+	const fieldsInReturnValue = _fieldsInReturnValue ?? {};
+
+	const fields = getFieldsFromCallback({
+		fields: fieldsInReturnValue,
+		callbacks: more,
+	});
+
+	const {
+		reader: readerInstance,
+		contentLength,
+		name,
+		contentType,
+		supportsContentRange,
+	} = await readerInterface.read({src, range: null, signal});
+
+	if (contentLength === null) {
+		throw new Error(
+			'Cannot read media without a content length. This is currently not supported. Ensure the media has a "Content-Length" HTTP header.',
+		);
+	}
```

Comment:
@JonnyBurger 

I notice that this breaks support for data urls used as src in `<Audio>`